### PR TITLE
Add anagrams in strings

### PR DESCRIPTION
**Pull request by ReneganRonin**

### CODE REVIEW

File: src/strings/detect_anagrams.jl [hunk 1]
```diff
@@ -0,0 +1,9 @@
+"""
+    detect_anagrams(subject, candidates)
+A function that checks if a list of words
+is an anagram or not of a subject word.
```
Comment (Panquesito7) [inline src/strings/detect_anagrams.jl:4]: Please add a Wikipedia reference of this algorithm.

Comment (Panquesito7) [inline src/strings/detect_anagrams.jl:4]:
```suggestion
is an anagram or not of a subject word.

Contributed By:- [Soc V. E.](https://github.com/ReneganRonin)
```

File: src/strings/detect_anagrams.jl [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+"""
+function detect_anagrams(subject, candidates)
```
Comment (Panquesito7) [inline src/strings/detect_anagrams.jl:6]: Please add documentation for this function (see, for example, the [`knapsack`](https://github.com/TheAlgorithms/Julia/blob/main/src/knapsack/knapsack.jl) problem).

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please fix the [Codecov](https://github.com/TheAlgorithms/Julia/pull/55/checks?check_run_id=2850291641) warnings.

File: src/strings/detect_anagrams.jl [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+A function that checks if a list of words
+is an anagram or not of a subject word.
+
+Contributed By:- [Soc V. E.](https://github.com/ReneganRonin)
+Based on my exercism's Julia track problem solution on [Anagrams](https://exercism.io/my/solutions/0a1d39faa5694624b7750be3ebf6c974).
+
+Instructions:
+
+An anagram is a rearrangement of letters to form a new word. Given a word and a list of candidates, select the sublist of anagrams of the given word.
+
```
Comment (Panquesito7) [inline src/strings/detect_anagrams.jl:12]:
```suggestion
```

File: src/strings/detect_anagrams.jl [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+Given "listen" and a list of candidates like "enlists" "google" "inlets" "banana" the program should return a list containing "inlets".
+
+Source
+
+Inspired by the Extreme Startup game https://github.com/rchatley/extreme_startup
```
Comment (Panquesito7) [inline src/strings/detect_anagrams.jl:17]:
```suggestion
Inspired by the [Extreme Startup game](https://github.com/rchatley/extreme_startup)
```

File: src/strings/detect_anagrams.jl [hunk 5]
```diff
@@ -0,0 +1,36 @@
+"""
+    detect_anagrams(subject, candidates)
+A function that checks if a list of words
+is an anagram or not of a subject word.
```
Comment (Panquesito7) [inline src/strings/detect_anagrams.jl:4]: Could you please add documentation for the function parameters?

Comment (Panquesito7) [inline src/strings/detect_anagrams.jl:4]:
```suggestion
is an [Anagram](https://en.wikipedia.org/wiki/Anagram) or not of a subject word.
```

**Review by Panquesito7 — COMMENTED** [review]
Almos there! 😄

**Review by Panquesito7 — APPROVED** [review]
LGTM. Thank you for your contribution! 👍 🎉

**Comment by Panquesito7** [comment]
@arubhardwaj, or @Azzaare, could you please do the code review? Thanks. 🙂

**Review by arubhardwaj — CHANGES_REQUESTED** [review]
Some CIs are conflicting, please resolve them!

**Review by Panquesito7 — COMMENTED** [review]
Any reasons why the [Codecov](https://github.com/TheAlgorithms/Julia/pull/55/checks?check_run_id=2923303166) checks are failing? @Azzaare @arubhardwaj.

**Comment by Azzaare** [comment]
I don't have time to check now, but my guess is that some part of the code are not covered by the test, thus decreasing the coverage.
Coverage bug might occur for some inline functions, but we can make the code coverage tool a bit more lenient.

For now, we can probably merge current PRs that succeed on CI build but not on coverage.

Once again, as long as we don't provide solid contribution guidelines (and we're late because of me ><) we should be kind of lenient

**Review by Azzaare — APPROVED** [review]

**Review by Azzaare — CHANGES_REQUESTED** [review]
@ReneganRonin you forgot to add the string test file in runtests.jl. Could you take care of it?

**Comment by Azzaare** [comment]
@ReneganRonin you're in for debugging now :p

**Comment by ReneganRonin** [comment]
@Panquesito7 Codecov issue fixed. it seems to be an issue from `is_palindrome` so i fixed it.

**Comment by Azzaare** [comment]
Thanks! Will try to merge things as soon as I am in front of a computer!

**Review by Azzaare — DISMISSED** [review]
Good to go

**Comment by ReneganRonin** [comment]
lol 😂 i didnt notice i add the commit verlet integration when i was doing demonstration for Ved. going to rebase and drop them lol

**Review by Azzaare — APPROVED** [review]

**Pull request merged**